frame/staking/src/lib.rs

```diff
@@ -2449,7 +2452,9 @@ impl<T: Config> Module<T> {
 			&ledger.stash,
 			validator_staking_payout + validator_commission_payout
 		) {
-			Self::deposit_event(RawEvent::Reward(ledger.stash, imbalance.peek()));
+            let imbalance_amount = imbalance.peek();
+            T::ValidatorReward::on_unbalanced(imbalance);
```

Comment:
what does this do?

Reply:
It calls the `on_unbalanced` method that provides a definition for this type. In our case, it will call `ValidatorsManager.PositiveImbalanceHandler.OnUnbalanced` which will balance the positive imbalance created by paying money to the validators and nominators